src/Microsoft.AspNet.OData.Shared/Query/ISelectExpandWrapper.cs

```diff
@@ -10,8 +10,14 @@ namespace Microsoft.AspNet.OData.Query
     /// <summary>
     /// Represents the result of a $select and $expand query operation.
     /// </summary>
-    public interface ISelectExpandWrapper
+    public interface ISelectExpandWrapper : IEdmEntityObject
     {
+        /// <summary>
+        /// The instance of the object that is being wrapped, if available.
+        /// </summary>
+        // The public contract should avoid propagating the "UntypedInstance" terminology.
+        object Instance { get; }
```

Comment:
>object Instance { get; } [](start = 8, length = 24)

This will be a breaking change if anyone has a custom implementation of ISelectExpandWrapper.